src/runs/execution/event-handlers/streaming.ts

```diff
@@ -334,7 +339,12 @@ export function createBeeStreamingHandler(ctx: AgentContext) {
 
 export function createStreamlitStreamingHandler(ctx: AgentContext) {
   const logger = getLogger().child({ runId: ctx.run.id, agent: Agent.STREAMLIT });
-  return (emitter: Emitter<StreamlitEvents>) => {
+  return async (emitter: Emitter<StreamlitEvents>) => {
```

Comment:
I tested the backend code from POSTMAN / UI and it worked. The `observe` implementation is simple:
```
observe(fn: (emitter: Emitter<GetRunInstance<I>>) => void) {
    fn(this.runContext.emitter);
    return this;
}
```

I can move the
```
if (emitter.trace?.id) {
      ctx.run.trace = new Trace({ id: emitter.trace.id });
      await ORM.em.flush();
    }
```
after `emitter.on` blocks to register them immediately.

Reply:
`observe(fn: (emitter: Emitter<GetRunInstance<I>>) => void) {
    fn(this.runContext.emitter);
    return this;
}`

This may result in unhandled promises and race conditions.

Reply:
Moving it after would technically resolve race conditions but it's still dangerous implementation. We should rather update the framework with a simple `await` if possible.

Reply:
Support for async middlewares/observes has been added in v0.0.51